tests-e2e/ocw-ci-test-course/video.spec.ts

```diff
@@ -0,0 +1,123 @@
+    name: `Download button`
+  })
+  await downloadButton.focus()
+  await page.keyboard.press("Enter")
+
+  const videoDownloadLink = page.getByRole("link", { name: "Download video" })
+  const transcriptDownloadLink = page.getByRole("link", {
+    name: "Download transcript"
+  })
+  const downloadLinksArr = [videoDownloadLink, transcriptDownloadLink]
+
+  for (let i = 0; i < 2; i++) {
+    await expect(downloadLinksArr[i]).toBeVisible()
+    await page.keyboard.press("Tab")
+    const hrefAttribute = await page.locator(":focus").getAttribute("href")
+    expect(hrefAttribute).toBe(downloadLinks[i])
+  }
+})
+
+test("Embed video redirects to video page using keyboard navigation", async ({
+  page
+}) => {
+  const course = new CoursePage(page, "course")
+  await course.goto("pages/video-series-overview/")
+  const videoRedirectLink = page.getByRole("link", {
+    name: "View video page"
+  })
+  await videoRedirectLink.focus()
+  page.keyboard.press("Enter")
+  await course.page.waitForNavigation()
+  expect(course.page.url()).toContain(
+    "resources/ocw_test_course_mit8_01f16_l01v01_360p"
+  )
+})
+test("Video tabs content (links) are keyoard navigable", async ({ page }) => {
+  const tabDict: { [key: string]: { title: string; redirectPath: string } } = {
+    "related-resource": {
+      title:        "Related Resources",
+      redirectPath: "courses/ocw-ci-test-course/resources/example_pdf/"
+    },
+    "optional-tab": {
+      title:        "Optional Tab",
+      redirectPath: "courses/ocw-ci-test-course/resources/example_notes/"
+    }
+  }
+  for (const tabClass in tabDict) {
+    if (tabClass) {
+      const course = new CoursePage(page, "course")
+      await course.goto("resources/ocw_test_course_mit8_01f16_l01v01_360p")
+      const tabTitle = tabDict[tabClass].title
+      const tabButton = page.getByRole("button", {
+        name: `${tabTitle}`
+      })
+      await tabButton.focus()
+      page.keyboard.press("Enter")
+      page.keyboard.press("Tab")
+      page.keyboard.press("Enter")
+      await course.page.waitForNavigation()
+      expect(course.page.url()).toContain(tabDict[tabClass].redirectPath)
+    }
+  }
+})
+test("Expand and collapse video tabs using keyboard navigation", async ({
+  page
+}) => {
+  const course = new CoursePage(page, "course")
+  await course.goto("resources/ocw_test_course_mit8_01f16_l01v01_360p")
+
+  const tabClassToTitle: { [key: string]: string } = {
+    transcript:         "Transcript",
+    "related-resource": "Related Resources",
+    "optional-tab":     "Optional Tab"
+  }
+  for (const tabClass in tabClassToTitle) {
+    if (tabClass) {
+      const title = tabClassToTitle[tabClass]
+      const tabButton = page.getByRole("button", {
+        name: `${title}`
+      })
+      await expect(tabButton).toBeVisible()
+      await tabButton.focus()
+
+      // Use keyboard to expand the tab
+      page.keyboard.press("Enter")
+
+      const toggleSection = await page.waitForSelector(
```

Comment:
@ibrahimjaved12 Carey's PR #1138 just added a `VideoElement` that would might be helpful here:

```ts
const course = new CoursePage(page, "course")
const video = new VideoElement(page) // "video" or "videoElement" or "videoSection", I dunno the best name

video.tab({ name: title, expanded: true }).waitFor()
await page.keyboard.press("Enter")
video.tab({ name: title, expanded: false }).waitFor()
```
Video is a ["Page Object Model"](https://playwright.dev/docs/pom).

One advantage of `video.tab(...)` over `page.getByRole("tab", ...) is that `video.tab` will scope itself to the video element. So if we end up adding other tabs elsewhere (e.g., the "Course Info" thing on right side of page should probably be a "tab"), our locators will still be video-specific.

Reply:
Done. Please let me know if that's how you meant.
Still had to use
```
    await expect(
      page.locator(`.video-tab.container.${tabClass}.collapse`)
    ).toHaveClass(/show/)
```
at one point though, otherwise I was getting this error:
`waiting for getByRole('tab', { name: 'Transcript', expanded: false }) to be visible`

Reply:
Ah, that's annoying. IMO, either the button should be disabled until its fully expanded, or you should be able to collapse it immediately after expansion begins. *That's what I'd expect based on react component libraries.* But, it's a bootstrap thing, so not really in our control.

I suppose if you wanted to (partly) use a role-based selector here, you could do
```ts
    await expect(
      videoPage.tabPanel().locator(`:scope.${tabClass}`)
    ).toHaveClass(/show/)
```

*The tab titles have role "tab", and the tab contents have role "tabpanel"; `:scope` is a pseudo-selector that basically lets an element query itself.*

But I dunno if that's really any better.

Reply:
Yeah. Thanks for sharing that approach!